core/domain/exp_domain.py

```diff
@@ -1037,11 +1037,10 @@ def validate(self, strict=False):
                         if (dest_state not in curr_queue and
                                 dest_state not in processed_state_names):
                             curr_queue.append(dest_state)
-
-            if excluded_state_is_bypassable:
-                raise utils.ValidationError(
-                    'Cannot make %s a checkpoint as it is bypassable'
-                    % state_name_to_exclude)
+                if excluded_state_is_bypassable:
```

Comment:
In this case maybe we should put excluded_state_is_bypassable in the outer for loop, just above the "while" statement.

Reply:
Done

Reply:
@seanlip  If we put excluded_state_is_bypassable in the outer for loop above the 'while' statement, then when `non_initial_checkpoint_state_names` contains only 1 element, the for loop would run only once, thus the validation error would not be raised the check would only run if the for loop runs again.

We can either raise the validation error inside the 'while' loop or just after the 'while' loop. What do you think?

Reply:
No, I meant the declaration of that variable (the `excluded_state_is_bypassable = False` line) since each iteration of the for loop should initialize it independently -- right?

Reply:
(To be clear, the previous code was otherwise fine, besides the above.)

Reply:
Ah right, done!